src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/response.tag

```diff
@@ -45,6 +51,31 @@
       </c:choose>:
   </div>
   <c:choose>
+    <c:when test="${questionWithResponses.feedbackParticipantCommentsOnResponsesAllowed}">
+      <button type="button" class="btn btn-default btn-xs icon-button pull-right show-frc-${addOrEdit}-form"
+              id="button_add_comment-${questionIndex}-${responseIndex}"
+              data-responseindex="${responseIndex}" data-qnindex="${questionIndex}" data-toggle="tooltip"
+              data-placement="top" title="<%=Const.Tooltips.COMMENT_ADD%>"
+              <c:if test="${data.preview or (not data.submittable)}"> disabled </c:if>>
+        <span class="glyphicon glyphicon-comment glyphicon-primary"></span>
+      </button>
+      <div class="${divClassType}">
+          ${response.submissionFormHtml}
+        <br>
+        <ul class="list-group" id="responseCommentTable-${questionIndex}-${responseIndex}"
+            style="${not empty response.feedbackParticipantCommentOnResponse ? 'margin-top:15px;': 'display:none'}">
+          <c:if test="${hasCommentOnResponse}">
+            <shared:feedbackResponseCommentRowForFeedbackParticipant frc="${response.feedbackParticipantCommentOnResponse}"
+                responseIndex="${response.responseIndx}" qnIndex="${questionWithResponses.question.qnIndx}" moderatedPersonEmail="${moderatedPersonEmail}"
+                isSessionOpenForSubmission="${isSessionOpenForSubmission}" isPreview="${data.preview}"
+                submittable="${data.submittable}" isModeration="${data.moderation}"/>
+          </c:if>
+
+          <shared:feedbackResponseCommentAddFormForFeedbackParticipant frc="${response.feedbackResponseCommentAdd}" responseIndex="${response.responseIndx}"
```

Comment:
If hasCommentOnResponse, you don't need the form right?

Reply:
Should hide it, when the existing comment gets deleted -> can show it.

Reply:
Form is already hidden. At any point of time only one form is shown, either edit (if exists) or add.